File: ios/Game On/Views/CreateGameView.swift
//
//  CreateGameView.swift
//  Game On!
//
//  Created by Fynn Herr on 27/01/2025.
//

import SwiftUI

struct CreateGameView: View {
    @EnvironmentObject var connectionManager: MPConnectionManager
    @EnvironmentObject var game: GameService
    @AppStorage("yourName") var yourName = "Fynn"

    @State private var newPlayers: [String] = []
    @State private var startGame: Bool = false
    @State private var navigateToLobby: Bool = false

    var body: some View {
        NavigationStack {
            ZStack {
                Color.bgCol.edgesIgnoringSafeArea(.all) // Background color
                
                VStack {
                    // Header
                    Text("\(connectionManager.myPeerId.displayName)'s Lobby")
                        .font(.largeTitle)
                        .fontWeight(.bold)
                        .foregroundColor(.white)
                        .padding(.top, 20)
                    
                    // Scrollable List for Players
                    ScrollView {
                        LazyVStack(spacing: 10) {
                            // Leader
                            playerRow(name: connectionManager.myPeerId.displayName, role: "Leader")
                            
                            // Joined Players
                            ForEach(newPlayers, id: \.self) { player in
                                playerRow(name: player, role: "Member")
                            }
                        }
                        .padding(.horizontal)
                    }
                    .scrollContentBackground(.hidden)
                    
                    Spacer()
                    
                    // Start Game Button
                    HStack {
                        Spacer()
                        Button("Start Game") {
                            // Start game action
                        }
                        .padding()
                        .background(Color.button, in: RoundedRectangle(cornerRadius: 10))
                        .font(.title2)
                        .fontWeight(.bold)
                        .foregroundColor(.white)
                    }
                    .padding()
                }
            }
            .colorScheme(.dark) // Dark mode
            .onAppear {
                connectionManager.startAdvertising()
                print("Start advertising...")
            }
            .onDisappear {
                connectionManager.stopAdvertising()
                connectionManager.resetConnectionState()
                print("Stop advertising...")
            }
            .onChange(of: connectionManager.paired) { paired in
                if paired {
                    navigateToLobby = true
                    //print("Connection established, navigating to LobbyView")
                }
            }
            /*.navigationDestination(isPresented: $navigateToLobby) {
                //LobbyView()
                CreateGameView()
            }*/
            .alert("Join Request", isPresented: $connectionManager.receivedInvite) {
                if let receivedPeer = connectionManager.receivedInviteFrom {
                    Button("Accept") {
                        newPlayers.append(receivedPeer.displayName)
                        connectionManager.invitationHandler?(true, connectionManager.session)
                    }
                    Button("Reject") {
                        connectionManager.invitationHandler?(false, nil)
                    }
                }
            } message: {
                Text("\(connectionManager.receivedInviteFrom?.displayName ?? "Unknown") wants to join your game.")
            }
        }
    }

    private func playerRow(name: String, role: String) -> some View {
        HStack {
            Text(name)
                .foregroundColor(.white)
            Spacer()
            Text(role)
                .foregroundColor(role == "Leader" ? .green : .blue)
        }
        .padding()
        .background(Color.gray.opacity(0.2))
        .cornerRadius(10)
    }
}

#Preview {
    CreateGameView()
        .environmentObject(MPConnectionManager(yourName: "Sample"))
        .environmentObject(GameService())
}

File: lib/views/CreateGameView.dart
import 'package:flutter/material.dart';
import '../main.dart';

class CreateGameView extends StatefulWidget {
  @override
  _CreateGameViewState createState() => _CreateGameViewState();
}

class _CreateGameViewState extends State<CreateGameView> {
  List<String> newPlayers = [];
  bool startGame = false;
  bool navigateToLobby = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.black, // Equivalent to Color.bgCol
      body: SafeArea(
        child: Column(
          children: [
            // Header
            Padding(
              padding: const EdgeInsets.only(top: 20.0),
              child: Text(
                "Your Name's Lobby", // Will be replaced with actual name
                style: TextStyle(
                  fontSize: 28,
                  fontWeight: FontWeight.bold,
                  color: Colors.white,
                ),
              ),
            ),
            
            // Scrollable List for Players
            Expanded(
              child: ListView(
                padding: EdgeInsets.symmetric(horizontal: 16),
                children: [
                  // Leader
                  playerRow(name: "Your Name", role: "Leader"), // Will be replaced with actual name
                  
                  // Joined Players
                  ...newPlayers.map((player) => playerRow(name: player, role: "Member")).toList(),
                ],
              ),
            ),
            
            // Start Game Button
            Padding(
              padding: const EdgeInsets.all(16.0),
              child: Align(
                alignment: Alignment.centerRight,
                child: ElevatedButton(
                  onPressed: () {
                    // Start game action
                  },
                  style: ElevatedButton.styleFrom(
                    backgroundColor: Colors.blue, // Equivalent to Color.button
                    padding: EdgeInsets.all(16),
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(10),
                    ),
                  ),
                  child: Text(
                    "Start Game",
                    style: TextStyle(
                      fontSize: 20,
                      fontWeight: FontWeight.bold,
                      color: Colors.white,
                    ),
                  ),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget playerRow({required String name, required String role}) {
    return Container(
      margin: EdgeInsets.symmetric(vertical: 5),
      padding: EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: Colors.grey.withOpacity(0.2),
        borderRadius: BorderRadius.circular(10),
      ),
      child: Row(
        children: [
          Text(
            name,
            style: TextStyle(color: Colors.white),
          ),
          Spacer(),
          Text(
            role,
            style: TextStyle(
              color: role == "Leader" ? Colors.green : Colors.blue,
            ),
          ),
        ],
      ),
    );
  }
}
